src/parser/Statement.ts

```diff
@@ -591,6 +591,37 @@ export class PrintStatement extends Statement {
     }
 }
 
+export class DimStatement extends Statement {
+    constructor(
+        public dimToken: Token,
+        public identifier?: Identifier,
+        public openingSquare?: Token,
+        public dimensions?: Expression[],
+        public closingSquare?: Token
+    ) {
+        super();
+        this.range = util.createRangeFromPositions(
+            this.dimToken.range.start,
+            (this.closingSquare ?? this.dimensions[this.dimensions.length - 1] ?? this.openingSquare ?? this.identifier ?? this.dimToken).range.end
+        );
+    }
+    public range: Range;
+
+    public transpile(state: TranspileState) {
+        const result = [
+            state.sourceNode(this.dimToken, 'dim'),
+            ' '
+        ];
+        return result;
+    }
+
+    public walk(visitor: WalkVisitor, options: WalkOptions) {
+        // if (this.expression && options.walkMode & InternalWalkMode.walkExpressions) {
```

Comment:
@elsassph can speak more to this, but I'm pretty sure we need to use the commented out if statement you have, and walk each of the dimensions. So something to the effect of 

```
 if (this.expression && options.walkMode & InternalWalkMode.walkExpressions) {
    for(let i = 0; i < this.dimensions.length; i++) {
        walk(this.dimensions, i, visitor, options, this);
    }
}
```